config/routes.rb

```diff
@@ -1,14 +1,18 @@
 Rails.application.routes.draw do
+  get 'shared_resources/create'
 
   resources :documents
   resources :reports
-  
+
   resources :projects do
     resources :documents
     resources :reports
   end
 
   resources :users, only: [:index]
+  resources :shared_resources, only: [:show, :create]
+
+  put 'shared_resources/:resource_type/:resource_id/:user_id', to: 'shared_resources#update_permission'
```

Comment:
This approach is interesting - any reason we couldn't just use a more generic update endpoint and only allow certain parameters? Is there a particular reason you chose this pattern instead?

Reply:
The only reason I did that was because I wanted to be more explicit about what this API does.
But I don't have a strong argument for or against any of those endpoints, I would be open to either. This would probably depend on who would be using this API. What would be the scenarios we would like to cover with Shared Resources.